### PR TITLE
Refactor parameters into config.js file

### PR DESCRIPTION
**Pull request by TimDaub**
TODOs:

- [x] Remove unnecessary parameters
- [x] Put all other parameters into a nice config
- [x] Let app bootstrap itself through config.json
- [x] Rebase against master
- [x] Go through `/src/components` and add all magic constants in config.js file

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+function findConfig(hostname) {
+  return configs.filter(({ DOMAINS }) => DOMAINS.includes(hostname));
+}
+
+export default function getConfig() {
+  const hostname = window.location.hostname;
+
+  const config = findConfig(hostname);
+  console.log(config);
```
Comment (TimDaub) [inline src/config.js]: Remove

File: src/components/Exchange.js [hunk 2]
```diff
@@ -922,53 +902,25 @@ export default class Exchange extends React.Component {
...
+                  // TODO: get real decimals
+                  const amount = bi(this.state.amount * 10 ** 18);
+                  const tokenAddr = this.props.daiContract._address;
+
+                  this.state.xdaiweb3.getColor(tokenAddr)
+                    .then(color =>
+                      Exit.fastSellAmount(
+                        this.state.daiAddress, amount, color,
+                        this.state.xdaiweb3, this.props.web3,
+                        'https://2nuxsb25he.execute-api.eu-west-1.amazonaws.com/testnet/sellExit'
```
Comment (troggy) [inline src/components/Exchange.js]: should use the MARKET_MAKER from CONFIG here?

File: src/config.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
+const configs = [
+  {
+    DOMAINS: ["localhost", "10.0.0.107", "sundai.fritz.box"],
+    SIDECHAIN: {
+      NAME: "Leap Testnet",
+      RPC: "http://theta.testnet.leapdao.org:8645",
```
Comment (troggy) [inline src/config.js]:
```suggestion
      RPC: "https://testnet-node.leapdao.org",
```

File: src/config.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      DAI_ADDRESS: "0xD2D0F8a6ADfF16C2098101087f9548465EC96C98",
+      MARKET_MAKER:
+        "https://2nuxsb25he.execute-api.eu-west-1.amazonaws.com/testnet",
+      BRIDGE_ADDRESS: "0x7DC67d56f5487D612985718221bD62D7b415Fc50",
+      EXPLORER: {
+        URL: "https://theta.leapdao.org/explorer/",
```
Comment (troggy) [inline src/config.js]:
```suggestion
        URL: "https://testnet.leapdao.org/explorer/",
```

File: src/config.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        NAME: "LeapDAO Theta Testnet Explorer"
```
Comment (troggy) [inline src/config.js]:
```suggestion
        NAME: "LeapDAO Testnet Explorer"
```

File: src/config.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+      UNISWAP: {
+        DAI_ETH_ADDRESS: "0x09cabec1ead1c0ba254b09efb3ee13841712be14"
+      }
+    }
+  },
+  {
+    DOMAINS: ["burner.leapdao.org"],
+    SIDECHAIN: {
+      NAME: "Leap Testnet",
+      RPC: "http://theta.testnet.leapdao.org:8645",
```
Comment (troggy) [inline src/config.js]:
```suggestion
      RPC: "https://testnet-node.leapdao.org",
```

File: src/config.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+      DAI_ADDRESS: "0xD2D0F8a6ADfF16C2098101087f9548465EC96C98",
+      MARKET_MAKER:
+        "https://2nuxsb25he.execute-api.eu-west-1.amazonaws.com/testnet",
+      BRIDGE_ADDRESS: "0x7DC67d56f5487D612985718221bD62D7b415Fc50",
+      EXPLORER: {
+        URL: "https://theta.leapdao.org/explorer/",
```
Comment (troggy) [inline src/config.js]:
```suggestion
        URL: "https://testnet.leapdao.org/explorer/",
```

File: src/config.js [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        NAME: "LeapDAO Theta Testnet Explorer"
```
Comment (troggy) [inline src/config.js]:
```suggestion
        NAME: "LeapDAO Testnet Explorer"
```

**Comment by TimDaub** [comment]
> Main question: what happened to all these `if (ERC20TOKEN) {` conditions and the logic they were guarding?

@johannbarbie Told me that the reason for `ERC20TOKEN` etc. was that Austin ran the wallet on large events where he didn't have the money to upfront to fill up that many paper wallets (e.g. 5000 people * 5€ = 25000€). So instead he used a worthless ERC20 token and promised event-caterers to be paid in relation to the amount of ERC20 tokens they earned after the event.

I removed this mainly for one reason: LeapDAO already found an easier way to do this with sunDAI, as we kept the rootchain DAI in the UI and allowed event participants to buy more sunDAI with DAI.

**Comment by johannbarbie** [comment]
> so no need to keep ERC20TOKEN around. Or am I missing something?

agree, we can do it on root chain like with sunDAI

**Pull request merged**